design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,239 @@
+We suspect (and hope) that this newer CRD versions functionality will encourage CRD authors to adopt a more strict process of updating their CRD versions, which would avoid the issue raised in this section.
+
+## Permissions and Privileges
+
+### Stack Manager
+
+Currently, the Stack Manager that installs stacks into the cluster runs with a `ClusterRoleBinding` to the cluster-admin `ClusterRole` because it needs to create RBAC permissions for the arbitrary CRDs that are owned and depended on by arbitrary stacks.
+A service account cannot grant RBAC permissions that it doesn't already have itself, so to support the arbitrary permissions needs of all stacks, we are currently using the cluster-admin role.
+
+This is a similar problem to what Helm's Tiller has historically had and we can consider revisiting this in the future.
+One possible solution would be to do more client side processing of stacks (instead of in-cluster processing) using the credentials and permissions of the interactive user,
+which would be similar to [Helm 3's](https://github.com/helm/community/blob/master/helm-v3/000-helm-v3.md) major design change.
+
+However, because we are limiting the [allowed set of resources](#allowed-resource-access) that a stack can have access to, this in-cluster Stack Manager model may be acceptable.
+Furthermore, moving to a client side model may limit the full lifecycle management abilities available from an in-cluster Stack Manager, such as automatic stack upgrades.
+
+A full exploration of this topic is out of scope for this version of the design document.
+
+### Stack Package Author
+
+The author of a stack will need to specify the intended scope of permissions for their stack.
+This should be known at stack build time because the stack's controller(s) will need to know whether to watch resources in a single namespace or across all namespaces.
+
+The `app.yaml` schema should be updated with a new field that captures the stack author's intended permission scope:
+
+```yaml
+# Human readable title of application.
+title: Sample Crossplane Stack
+
+# Scope of permissions needed by the stack once installed in the control plane,
+# current supported values are:
+#
+# - Cluster
+# - Namespace
+permissionScope: Cluster
+```
+
+### Stack Install
+
+Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to allow differing access to request their installation.
+A simple way to approach this would be to create different installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+Currently, we use a single `ExtensionRequest` type, but going forward we should differentiate by stack type:
```

Comment:
Well, now it's `StackRequest` : )

Reply:
Yes, I'll update this lingering ExtensionRequest reference